### PR TITLE
Compatibility changes for PHP 5.4

### PR DESCRIPTION
**Pull request by mccottry**
**WIP until actually tested in the PHP 5.4 context**

# Description
This changes code to be compatible with PHP 5.4 as we advertise that we support it

Fixes: https://boltpay.slack.com/archives/CKBBANKD2/p1594637919044000

#changelog Compatibility changes for PHP 5.4

# Type of change

- [x] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Test

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

**Comment by isklyarenko** [comment]
@mccottry thank you for this PR!
one more place is lib/Boltpay/GuzzleHttp/Exception/RequestException.php, lines 84 and 87 where ::class keyword is used. It causes a parse error. Could you fix it as well?

**Comment by mccottry** [comment]
> @mccottry thank you for this PR!
> one more place is lib/Boltpay/GuzzleHttp/Exception/RequestException.php, lines 84 and 87 where ::class keyword is used. It causes a parse error. Could you fix it as well?

@isklyarenko Please try these updated changes and let me know if there are any more incompatibilities.

**Comment by isklyarenko** [comment]
app/code/community/Bolt/Boltpay/Model/Order.php, line 867 brings an error 

`Can't use function return value in write context`

and need to be changed to something like this

```
$orderCreatedAt = $order->getCreatedAt();
if (empty($orderCreatedAt)) { $order->setCreatedAt(Mage::getModel('core/date')->gmtDate())->save(); }
```

**Comment by mccottry** [comment]
@isklyarenko 
I will create a dev instance that is using PHP 5.4 as there might be many unexpected things and going back and forward like this will take to long.

I hope to have a 5.4 tested release available by end-of-day Monday.

File: lib/Boltpay/GuzzleHttp/Handler/CurlFactory.php [hunk 1]
```diff
@@ -66,6 +66,18 @@ public function release(EasyHandle $easy)
...
+            /////////////////////////////////////////////////
+            // PHP 5.4 support
+            /////////////////////////////////////////////////
+            if (!function_exists('curl_reset'))
+            {
+                function curl_reset(&$ch)
+                {
+                    $ch = curl_init();
+                }
+            }
```
[the remark this answers is not in the document]
Comment (mccottry) [reply lib/Boltpay/GuzzleHttp/Handler/CurlFactory.php:78]: Actually, this is a namespacing issue.  It can be fixed by bracketed syntax which I will do.

**Comment by mccottry** [comment]
> @isklyarenko
> I will create a dev instance that is using PHP 5.4 as there might be many unexpected things and going back and forward like this will take to long.
> 
> I hope to have a 5.4 tested release available by end-of-day Monday.

@isklyarenko 
With that being said, I haven't tested any of these changes, but you can test them ahead of me in the context of the store which you are working on.  If you are able to do this and see success, please let us know.   Additionally, I will be able to test this on Monday.  Thank you for your thoroughness.

**Comment by isklyarenko** [comment]
> @isklyarenko
> With that being said, I have made changes based on your results, but haven't tested any of these changes. However, you can test them ahead of me in the context of the store which you are working on. If you are able to do this and see success, please let us know. Additionally, I will be able to test this on Monday. Thank you for your thoroughness.

Thanks @mccottry . I've tested everything on the merchant's test env and it looks good so far. I also asked them to test on their own as well

**Comment by adentong** [comment]
@isklyarenko Let me know how the testing goes, and we can merge this PR if all goes well.

**Comment by isklyarenko** [comment]
@adentong @mccottry didn't get any update from the merchant so far, but it looks good to me and I think we can merge it.

@mccottry PR still has a label WIP. Are you going to change anything else?

**Comment by adentong** [comment]
Since this appears to be tested I'm going to remove the WIP label

**Pull request merged**